source/particle/world.cc

```diff
@@ -1285,32 +1285,71 @@ namespace aspect
 
       const FiniteElement<dim> &velocity_fe = this->get_fe().base_element(this->introspection().base_elements.velocities);
 
+      const bool compute_fluid_velocity = this->include_melt_transport() && property_manager->get_data_info().fieldname_exists("melt_particle");
+      unsigned int fluid_component_index;
+      std::vector<bool> use_solid_velocity;
+      if (compute_fluid_velocity)
+        {
+          fluid_component_index = this->introspection().variable("fluid velocity").first_component_index;
+          const unsigned int melt_property_index = property_manager->get_property_component_by_name("melt_particle");
+
+          use_solid_velocity.resize(particles_in_cell,true);
+
+          typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
+          for (unsigned int particle_index = 0; it!=end_particle; ++it,++particle_index)
+            if (*it->get_properties()[melt_property_index] == 1.0)
```

Comment:
I haven't read further down in the patch yet, but does a value of `1.0` indicate that you want to advect the particle with the melt velocity? Is this property never changed? Because if you do, then comparing by equality two floating point numbers may be troublesome.